packages/server/src/automations/steps/ai/extract.ts

```diff
@@ -0,0 +1,94 @@
+import { ai } from "@budibase/pro"
+import * as automationUtils from "../../automationUtils"
+import {
+  DocumentSourceType,
+  ExtractFileDataStepInputs,
+  ExtractFileDataStepOutputs,
+} from "@budibase/types"
+import { objectStore } from "@budibase/backend-core"
+import { Readable } from "stream"
+import fetch from "node-fetch"
+
+export async function run({
+  inputs,
+}: {
+  inputs: ExtractFileDataStepInputs
+}): Promise<ExtractFileDataStepOutputs> {
+  if (!inputs.file || !inputs.schema) {
+    return {
+      success: false,
+      data: {},
+      response:
+        "Extract Document Data AI Step Failed: File and Schema are required.",
+    }
+  }
+
+  try {
+    const llm = await ai.getLLMOrThrow()
+
+    let filename
+    let contentType
+    let fileIdOrDataUrl
+    let request
+
+    if (
```

Comment:
For readability and verbosity, on long functions like this would be good to break some logic into subfunctions